tests/testthat/test-derive_var_confirmation_flag.R

```diff
@@ -0,0 +1,269 @@
+  "2",      1,        "SD",
+  "2",      2,        "PR",
+  "2",      3,        "PD",
+  "3",      1,        "SD",
+  "4",      1,        "PR",
+  "4",      2,        "PD",
+  "4",      3,        "SD",
+  "4",      4,        "SD",
+  "4",      5,        "PR"
+)
+
+# derive_var_confirmation_flag ----
+## Test 1: filter without first_cond ----
+## Flagging any patient PR value that is followed subsequently by a CR or PR
+test_that("derive_var_confirmation_flag Test 1: filter without first_cond", {
+  actual <-
+    derive_var_confirmation_flag(
+      data,
+      new_var = CONFFL,
+      by_vars = vars(USUBJID),
+      join_vars = vars(AVISITN, AVALC),
+      join_type = "after",
+      order = vars(AVISITN),
+      filter = AVALC == "PR" & AVALC.join %in% c("CR", "PR")
+    )
+
+  expected <- tribble(
+    ~USUBJID, ~AVISITN, ~AVALC, ~CONFFL,
+    "1",      1,        "PR",   "Y",
+    "1",      2,        "CR",   NA_character_,
+    "1",      3,        "CR",   NA_character_,
+    "1",      4,        "SD",   NA_character_,
+    "1",      5,        "NE",   NA_character_,
+    "2",      1,        "SD",   NA_character_,
+    "2",      2,        "PR",   NA_character_,
+    "2",      3,        "PD",   NA_character_,
+    "3",      1,        "SD",   NA_character_,
+    "4",      1,        "PR",   "Y",
+    "4",      2,        "PD",   NA_character_,
+    "4",      3,        "SD",   NA_character_,
+    "4",      4,        "SD",   NA_character_,
+    "4",      5,        "PR",   NA_character_
+  )
+
+  expect_dfs_equal(
+    base = expected,
+    compare = actual,
+    keys = c("USUBJID", "AVISITN")
+  )
+})
+
+## Test 2: filter with first_cond ----
+## Flagging any patient CR value that is followed subsequently by a CR
+test_that("derive_var_confirmation_flag Test 2: filter with first_cond", {
+  actual <-
+    derive_var_confirmation_flag(
+      data,
+      new_var = CONFFL,
+      by_vars = vars(USUBJID),
+      join_vars = vars(AVALC),
+      first_cond = AVALC == "CR" &
+        AVALC.join == "CR",
+      order = vars(AVISITN),
+      filter = TRUE
+    )
+
+  expected <- tribble(
+    ~USUBJID, ~AVISITN, ~AVALC, ~CONFFL,
+    "1",      1,        "PR",   NA_character_,
+    "1",      2,        "CR",   "Y",
+    "1",      3,        "CR",   NA_character_,
+    "1",      4,        "SD",   NA_character_,
+    "1",      5,        "NE",   NA_character_,
+    "2",      1,        "SD",   NA_character_,
+    "2",      2,        "PR",   NA_character_,
+    "2",      3,        "PD",   NA_character_,
+    "3",      1,        "SD",   NA_character_,
+    "4",      1,        "PR",   NA_character_,
+    "4",      2,        "PD",   NA_character_,
+    "4",      3,        "SD",   NA_character_,
+    "4",      4,        "SD",   NA_character_,
+    "4",      5,        "PR",   NA_character_
+  )
+
+  expect_dfs_equal(
+    base = expected,
+    compare = actual,
+    keys = c("USUBJID", "AVISITN")
+  )
+})
+
+## Test 3: filter with first_cond and summary function ----
```

Comment:
Same comment for Test 3 as already made for Test 2: Use of `subsequently` and setting `join_type = "after"`. Adding the extra test requested for test 2 not necessary for Test 3, unless you really want to!

Reply:
Removed "subsequently"

`join_type = after` was used implicitly when `first_cond` was used. As this behaviour is confusing, I have removed the implicit conditions and set the default for `join_type` to `"after"` (this is the more common use case).